app/src/main/java/com/ably/tracking/demo/publisher/common/ActivityLifecycleCallbacksAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.ably.tracking.demo.publisher.common
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+
+open class ActivityLifecycleCallbacksAdapter : Application.ActivityLifecycleCallbacks {
```

Comment:
Can I ask what's the purpose of this class is ?

Reply:
`Application.ActivityLifecycleCallbacks` is defined in the Android framework as an interface, so in order to implement it, we need to override all methods defined in it. In our case we care only about two of them and this Adapter class lets us override only the one we want in "feature" code making it more readable